fatrace.c

```diff
@@ -251,6 +253,12 @@ print_event (const struct fanotify_event_metadata *data,
                 len--;
             procname[len] = '\0';
             procname_pid = data->pid;
+
+            /* get user and group */
+            if (option_user == 1) {
+                if (fstat (proc_fd, &ug) < 0)
```

Comment:
I found this to work rather poorly. The `comm` file is often owned by root:root, especially with `touch` and `rm`, i.e. short-running processes. Stating /proc/pid seems to work better.

Reply:
I reworked this to get the owner from /proc/pid, and using openat() to get `comm` in that.